config/initializers/sentry.rb

```diff
@@ -13,6 +13,17 @@ def filter_record_not_unique_exception_messages!(event, hint)
   end
 end
 
+def ignore_sidekiq_errors_if_retry(event)
+  # Check if the Sidekiq job is still retryable
+  if event&.exception.is_a?(Faraday::Error) &&
+       (
+         event&.transaction&.context&.job.is_a?(ActiveStorage::AnalyzeJob) ||
+           event&.transaction&.context&.job.is_a?(ActiveStorage::PurgeJob)
+       ) && event&.transaction&.context&.job&.retry?
```

Comment:
Could we assign `event.transaction.context.job` to a variable so it's easier to read, perhaps something like:

```rb
return unless event&.exception.is_a?(Faraday::Error)

job = event&.transaction&.context&.job
return if job.nil?

if (job.is_a?(ActiveStorage::AnalyzeJob) || job.is_a?(ActiveStorage::PurgeJob) && job.retry?
  ...
end
```